### ISSUE TITLE
Støtte for program-feltnavn i programmer

### ISSUE THREAD

**Issue by tarjeieo** [Puzzlepart/prosjektportalen365#1225]
### Er forespørselen din relatert til et problem? Vennligst beskriv.

Per i dag har programmalen sitt eget oppsett med egne faser, egen forside og egen venstremeny. Feltnavnene i prosjektinformasjon bruker de samme som prosjekter: Prosjektleder, prosjekteier osv.

### Beskriv løsningen du ønsker

Implementer støtte (om de ikke allerede er det) for å overskrive titlene på felter  i prosjektegenskaper slik at vi kan bruke begrepene som matcher bedre et program... "Programleder", "Programeier" osv... Det er nok også behov for å overskrive beskrivelsene på noen felter.

Ser det kan være utfordrende mtp. synkronisering av felter mot sentral liste, så vi kan godt diskutere løsning for hvordan vi best oppnår dette - skal synkroniseringen for eksempel gjøre oppslag mot "oversettelser" fra malfilen eller egen liste?

**Comment by tarjeieo** [comment]
Vi løser det med en egen liste med alternative labels, med verdier for navn og beskrivelse, koblet mot prosjekt-innholdstype eller prosjektmal.

Dette vil da kunne benyttes i implementeringen av #1227

**Comment by olemp** [comment]
@tarjeieo @Remi749 Mitt forslag er ett nytt felt på "Prosjetkolonne"-innholdstypen (f.eks. `GtFieldOverrides` med visningsnavn `Tilpassede feltinnstillinger`.

Eksempel for "Prosjektnavn i økonomisystemet":

```json
{
  "program": {
    "nb-no": {
      "displayName": "Programnavn i økonomisystemet",
      "description": "Hvis relevant, legg inn tilhørende programnavn fra økonomisystemet."
    }
  }
}
```

Da kan vi i i f.eks. Prosjektinformasjon velge å bruke `program` "konfigurasjonen".

<img width="984" alt="image" src="https://github.com/Puzzlepart/prosjektportalen365/assets/7606007/80c3d26d-0a28-4f53-88a3-dd41cff79a17">


<img width="421" alt="image" src="https://github.com/Puzzlepart/prosjektportalen365/assets/7606007/5937827f-9c6e-40be-8902-32b7c57fe62d">

**Comment by tarjeieo** [comment]
Det er en god idé - skulle vi bundet det til malen istedenfor, eller ser du for deg at det kan være greit å ha konfigurasjoner frikoblet fra malene? Jeg tror egentlig det er greit å ha de frikoblet, men kunne vært fornuftig om malene bestemte hvilken feltkonfigurasjon de skal bruke? Da kan man lage Programmal, "Østvest kommunes programmal", "Store programmer" osv. som alle er kobla til "program"-konfigurasjonen

**Comment by olemp** [comment]
> Det er en god idé - skulle vi bundet det til malen istedenfor, eller ser du for deg at det kan være greit å ha konfigurasjoner frikoblet fra malene? Jeg tror egentlig det er greit å ha de frikoblet, men kunne vært fornuftig om malene bestemte hvilken feltkonfigurasjon de skal bruke? Da kan man lage Programmal, "Østvest kommunes programmal", "Store programmer" osv. som alle er kobla til "program"-konfigurasjonen

Tenker det er bedre å ha selve konfigurasjonen på prosjektkolonnene. Da kan vi også støtte å vise program-feltnavn i porteføljevisninger som skal vise kun programmer. Malene kan da peke på konfigurasjonen som skal brukes, f.eks. `program`.

**Comment by tarjeieo** [comment]
Ja, veldig med deg på den! Da blir det en kolonne på malen for "Feltkonfigurasjon" el.l.?

**Comment by olemp** [comment]
> Ja, veldig med deg på den! Da blir det en kolonne på maloppsett for "Feltkonfigurasjon" el.l.?

Hvilke felter tenker du at skal ha egne navn/beskrivelser for program?

**Comment by olemp** [comment]
> > Ja, veldig med deg på den! Da blir det en kolonne på maloppsett for "Feltkonfigurasjon" el.l.?
> 
> Hvilke felter tenker du at skal ha egne navn/beskrivelser for program?

Og hva vil du ha som feltnavn og beskrivelse for disse @tarjeieo?

**Comment by olemp** [comment]
> > > Ja, veldig med deg på den! Da blir det en kolonne på maloppsett for "Feltkonfigurasjon" el.l.?
> > 
> > 
> > Hvilke felter tenker du at skal ha egne navn/beskrivelser for program?
> 
> Og hva vil du ha som feltnavn og beskrivelse for disse @tarjeieo?
> 
> Legg gjerne inn en tabell som dette i beskrivelsen av dette issuet.
> 
> #	Felt/kolonne	Visningsnavn	Beskrivelse
> 1	Prosjektnavn i økonomisystemet	?	?

Da er støtten lagt inn gjennom #1234, men det er ikke konfigurert opp noen kolonner enda.

**Comment by tarjeieo** [comment]
| #   | Felt/kolonne          | Visningsnavn     | Beskrivelse |
| --- | ----------------- | ---------------------- |---------------------- |
|1|Prosjektnavn i økonomisystemet|Navn i økonomisystemet|Hvis relevant, legg inn tilhørende program/prosjektnavn fra økonomisystemet|
|2|Prosjektnummer|Program/prosjektnummer|Hvis relevant, legg inn tilhørende program/prosjektnummer|
|2|Tjenesteområde|Tjenesteområde|Hva slags tjenesteområde hører programmet til?|
|2|Prosjekttype|Programtype|Hva slags type program er dette?|
|2|Prosjektleder|Programleder|Hvem er leder for programmet?|
|2|Prosjekteier|Programeier|Hvem er eier av programmet?|
|2|Prosjektstøtte|Støtteressurser|Hvem er støtteressurser for programmet? Dette kan være assisterende program/prosjektledere eller andre involverte.|
|2|Gevinstansvarlig|Gevinstansvarlig|Hvem er gevinstansvarlig for programmet?|
|2|Målsetting|Målsetting|Hva er programmets målsetting?|
|2|Prosjektstatus|Programstatus|Hvilken status har programmet?|
|2|Prosjektadministrasjonsroller|Prosjektadministrasjonsroller|Velg hvilke roller som skal kunne administrere programmet. Dette inkluderer handlinger som å redigere egenskaper, fase og status.|

Tar gjerne innspill fra @pzljanb på dette også - mulig programfelter uansett bør ha en egen innholdstype med andre felter, men dette blir iallefall mer riktig enn det vi har i dag.

**Comment by olemp** [comment]
@tarjeieo Vi får ikke oppdatert kolonnene slik det er nå.

![image](https://github.com/Puzzlepart/prosjektportalen365/assets/7606007/26ea45e1-de0e-480a-ba57-ec315afa9fc3)

Vi kan endre til overwrite, men jeg vet ikke hvor mye justeringer brukerne av PP har gjort på "våre" kolonner?

**Comment by tarjeieo** [comment]
Ja, ser den. Må evt se om vi rekker å lage et oppgr skript som bare tar den nye feltverdien. Ellers kan vi leve med det for nå tenker jeg.

**Comment by olemp** [comment]
> Ja, ser den. Må evt se om vi rekker å lage et oppgr skript som bare tar den nye feltverdien. Ellers kan vi leve med det for nå tenker jeg.

Vi må ikke gjøre oppgraderingen for kompleks heller. Det som er problemet. Tipper kundene ikke gjøre allverdens med endringer på standardkolonner.

**Comment by olemp** [comment]
Da kan dette testes [her](https://puzzlepart.sharepoint.com/sites/OsloProgramfortjenesteinnovasjon/SitePages/ProgramHome.aspx), @tarjeieo. 

![image](https://github.com/Puzzlepart/prosjektportalen365/assets/7606007/e068c91b-eaee-4049-bb6b-7d8cae7fd7da)

**Comment by tarjeieo** [comment]
Fresh, ser ut til å fungere bra! :)

**Comment by olemp** [comment]
> Fresh, ser ut til å fungere bra! :)

Flott! Løst i #1314

**Comment by tarjeieo** [comment]
Testa denne i en splitter ny 1.9.0 release og den fungerer dessverre ikke. Vi må sjekke hva som har skjedd. Program-verdiene kommer ikke, det er bare standard prosjekt-navn som vises.

**Comment by olemp** [comment]
> Testa denne i en splitter ny 1.9.0 release og den fungerer dessverre ikke. Vi må sjekke hva som har skjedd. Program-verdiene kommer ikke, det er bare standard prosjekt-navn som vises.

Er feltkonfigurasjonsfeltet satt til "Program" for Program-malen i Maloppsett?

**Comment by tarjeieo** [comment]
Ja, da fungerte det igjen - så vi må sette dette som default i upgrade.

**Comment by Remi749** [comment]
Dette ser ut til å være i orden nå, ved oppgradering settes `Program` verdi i `Feltkonfigurasjon` for `Programmal`
